homework/index.js

```diff
@@ -28,26 +22,139 @@
     });
     return elem;
   }
+  function formatDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
+
+  function renderError(error) {
+    const container = document.getElementById('main-container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'alert',
+    });
+  }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addRow(table, body, value) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, { text: body });
+    createAndAppend('td', tr, { text: value });
+    return tr;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
-        });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const table = createAndAppend('table', leftSide);
+    const tr1 = addRow(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
     });
+    addRow(table, 'Description:', repo.description);
+    addRow(table, 'Fork: ', repo.forks);
+    addRow(table, 'Updated:', formatDate(repo.updated_at));
   }
 
+  function renderContributor(contributor, contributorList) {
+    const li = createAndAppend('li', contributorList);
+    const list = createAndAppend('a', li, {
```

Comment:
You are creating an anchor tag here, not a list. Your variable name should reflect that.